packages/svelte/src/compiler/phases/3-transform/server/transform-server.js

```diff
@@ -423,7 +423,10 @@ const global_visitors = {
 		}
 
 		if (rune === '$state.snapshot') {
-			return /** @type {import('estree').Expression} */ (context.visit(node.arguments[0]));
+			return b.call(
+				'structuredClone',
```

Comment:
gah, we can't just do this on the server because of this case:

```js
class Foo {
  x = $state()

  toJSON() {
    return { x: this.x };
  }
}
```

With `structuredClone` that would become `{}` rather than `{ x: 1 }`.

Is there any way to influence what `structuredClone` 'sees' when encountering an object like this?

Reply:
Addressed this by using the same function on server as on client